src/helics/core/fileConnections.hpp

```diff
@@ -48,9 +48,61 @@ namespace fileops {
                         });
                     } else {
                         std::string ipt = getOrDefault(conn, "input", std::string());
-                        addTargets(conn, "targets", [brk, &ipt](const std::string& target) {
-                            brk->dataLink(target, ipt);
+                        if (!ipt.empty()) {
+                            addTargets(conn, "targets", [brk, &ipt](const std::string& target) {
+                                brk->dataLink(target, ipt);
+                            });
+                            addTargets(conn, "sources", [brk, &ipt](const std::string& target) {
+                                brk->dataLink(target, ipt);
```

Comment:
It seems like the various dataLink functions arguments might be better named `publication` and `input` in a later PR.

Is a data link created from the same publication/input getting added to both "targets" and "sources" because the created link is bidirectional?

```suggestion
                            addTargets(conn, "targets", [brk, &ipt](const std::string& pub) {
                                brk->dataLink(pub, ipt);
                            });
                            addTargets(conn, "sources", [brk, &ipt](const std::string& pub) {
                                brk->dataLink(pub, ipt);
```